Update site identification step Heading, SubTitle and CTA text

Imran92 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/91477, https://github.com/Automattic/wp-calypso/issues/91478, https://github.com/Automattic/wp-calypso/issues/91542

## Proposed Changes

We've updated the -
- Heading of the site identification page from "`Let’s import your content`" to "`Let’s find your site`"
- The subtitle from "`Drop your current site address below to get started.`" to "`Drop your current site address below to get started. In the next step, we'll measure your site's performance and confirm its eligibility for migration.`"
- The CTA from "Continue" to "Check my site"

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* Previously we only said 'what', now we are also adding the 'why' to explain the need and benefit of putting the URL in the input field

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- First, switch to `trunk`
- Go to `https://wordpress.com/me/account`
- Switch to any language other than English
- In the migration flow (take any migration path, for example `http://calypso.localhost:3000/start`), navigate to the Site Identification step (screenshot attached below in case you're not sure which page)
- Check the header, subtitle, and CTA text on that page (screenshot is marked to point out which parts)
- Copy-paste them somewhere so that you can verify them later
- Now switch to this branch, and after rebuilding, open the site identification tab and reload
- Make sure all of those three texts are still looking the same as before in your selected language (you can verify using your copy)
- Now go to the `https://wordpress.com/me/account` page and switch the language to English.
- Get back to the  Site Identification page
- Make sure you see the updated texts in English
 
<img width="1045" alt="Screenshot 2024-06-06 at 6 47 03 AM" src="https://github.com/Automattic/wp-calypso/assets/6820724/f215abef-51ad-41a7-a0a1-ec78c8079584">

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Imran92 committed:
Update site identification step subtitle

merkushin commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-identify/index.tsx:
```diff
@@ -23,8 +25,20 @@ interface Props {
+	const newSubtitle = translate(
+		"Drop your current site address below to get started. In the next step, we'll measure your site's performance and confirm its eligibility for migration."
+	);
+	const isTranslationAvailableForNewSubtitle = hasTranslation(
```
I remember we shouldn't pass variables to the `translate()` function, is there a similar rule for `hasTranslation()`?

Imran92 replied:
I don't know for sure. But in this case I've repeated the string because even if we passed a variable to hasTranslation, we'd have to duplicate anyway

merkushin replied:
Ah, makes sense. We don't want to pass the translation here.
Thanks for clarification!

merkushin commented:
Tested it with Spanish. Works as described.

### Trunk
![CleanShot 2024-06-04 at 23 26 34@2x](https://github.com/Automattic/wp-calypso/assets/329356/73a85941-e247-4bfe-80ed-9fb04020156f)
![CleanShot 2024-06-04 at 23 28 06@2x](https://github.com/Automattic/wp-calypso/assets/329356/bb420301-9fca-4a89-b345-3895c6c6a206)

### Current branch
![CleanShot 2024-06-04 at 23 32 07@2x](https://github.com/Automattic/wp-calypso/assets/329356/04a10cfa-d12b-484f-8dc2-a42d2bac15fd)
![CleanShot 2024-06-04 at 23 31 25@2x](https://github.com/Automattic/wp-calypso/assets/329356/5eb068d7-291f-4911-a9f1-eafb88f6352d)

merkushin commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-identify/index.tsx:
```diff
@@ -23,8 +25,20 @@ interface Props {
+	const isEnglishLocale = useIsEnglishLocale();
 	const [ siteURL, setSiteURL ] = useState< string >( '' );
 
+	// TODO: Remove extra steps for non-English locales once we have translations.
```
Should we create an issue for that or just leave as a TODO and remove it "eventually"?

Imran92 replied:
Donna and I chatted a bit about it and we also discussed in our team meeting. Probably creating issue will create too much noise and be a bit overwhelming. As it's not very significant, a TODO seemed a milder way of indication

merkushin reviewed (approved):
👍

merkushin commented:
Do we need to add `[Status] String Freeze` label here?

renatho commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-identify/index.tsx:
```diff
@@ -23,8 +25,20 @@ interface Props {
+	const isTranslationAvailableForNewSubtitle = hasTranslation(
+		"Drop your current site address below to get started. In the next step, we'll measure your site's performance and confirm its eligibility for migration."
+	);
+	const subtitleInUse =
```
Nice! =)
It makes me think that at some point we could create a function on `@automattic/i18n-utils` to simplify this logic.

renatho replied:
Maybe I'll try to send a PR proposal this week.

Imran92 replied:
Marking it as resolved because we've decided to use the existing `useHasEnTranslation` instead

Imran92 committed:
Update title text

Imran92 committed:
Update email capture component to take button label

Imran92 committed:
use hasEnTranslation and change CTA text

Imran92 commented:
> Do we need to add `[Status] String Freeze` label here?

If we check for available translation for user's locale and show the translated version only when available, we don't need to put the label.

Imran92 committed:
Update tests

Imran92 committed:
Add new test for CTA label

Imran92 commented:
Asking for a review again as I've updated all strings in this PR instead of only the subtitle. Changed the translation hook that was used. Have also added and updated tests

merkushin commented:
Tested the updated version. Looks good 👍 

![CleanShot 2024-06-06 at 09 33 56@2x](https://github.com/Automattic/wp-calypso/assets/329356/727362f4-09f1-446a-abe8-433b8ab2831a)
![CleanShot 2024-06-06 at 09 34 33@2x](https://github.com/Automattic/wp-calypso/assets/329356/20742e1e-1b8b-440e-9303-c6a3706dbdf6)
![CleanShot 2024-06-06 at 09 37 19@2x](https://github.com/Automattic/wp-calypso/assets/329356/95835912-b118-4375-aff6-f80353c4aee5)
![CleanShot 2024-06-06 at 09 37 53@2x](https://github.com/Automattic/wp-calypso/assets/329356/37f51e15-74d9-4003-8e3b-d5b3dbbffcb0)

merkushin commented on client/blocks/import/capture/capture-input.tsx:
```diff
@@ -15,15 +15,16 @@ import type { OnInputChange, OnInputEnter } from './types';
+	dontHaveSiteAddressLabel?: string;
 	hasError?: boolean;
-	skipInitialChecking?: boolean;
+	hideImporterListLink?: boolean;
 	label?: ReactNode;
+	nextLabelText?: string;
+	onDontHaveSiteAddressClick?: () => void;
+	onInputChange?: OnInputChange;
+	onInputEnter: OnInputEnter;
 	placeholder?: string;
-	dontHaveSiteAddressLabel?: string;
-	hideImporterListLink?: boolean;
+	skipInitialChecking?: boolean;
```
It took some time for me to find the difference 😅, so I sum up for anyone who reads later:
- properties ordered alphabetically,
- `nextLabelText` property added.

merkushin reviewed (approved):
LGTM 👍

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/14177438

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @Imran92 for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/14177438).
